pkg/attribute/tracker_test.go

```diff
@@ -142,113 +135,79 @@ func TestTracker_ApplyAttributes(t *testing.T) {
 }
 
 func compareBags(b1 Bag, b2 Bag) bool {
-	if len(b1.StringKeys()) != len(b2.StringKeys()) {
+	if len(b1.Names()) != len(b2.Names()) {
 		return false
 	}
 
-	for _, k := range b1.StringKeys() {
-		v1, _ := b1.String(k)
-		v2, _ := b2.String(k)
-		if v1 != v2 {
-			return false
-		}
-	}
-
-	if len(b1.Int64Keys()) != len(b2.Int64Keys()) {
-		return false
-	}
-
-	for _, k := range b1.Int64Keys() {
-		v1, _ := b1.Int64(k)
-		v2, _ := b2.Int64(k)
-		if v1 != v2 {
-			return false
-		}
-	}
-
-	if len(b1.Float64Keys()) != len(b2.Float64Keys()) {
-		return false
-	}
-
-	for _, k := range b1.Float64Keys() {
-		v1, _ := b1.Float64(k)
-		v2, _ := b2.Float64(k)
-		if v1 != v2 {
-			return false
-		}
-	}
-
-	if len(b1.BoolKeys()) != len(b2.BoolKeys()) {
-		return false
-	}
-
-	for _, k := range b1.BoolKeys() {
-		v1, _ := b1.Bool(k)
-		v2, _ := b2.Bool(k)
-		if v1 != v2 {
-			return false
-		}
-	}
-
-	if len(b1.TimeKeys()) != len(b2.TimeKeys()) {
-		return false
-	}
-
-	for _, k := range b1.TimeKeys() {
-		v1, _ := b1.Time(k)
-		v2, _ := b2.Time(k)
-		if v1 != v2 {
-			return false
-		}
-	}
-
-	if len(b1.DurationKeys()) != len(b2.DurationKeys()) {
-		return false
-	}
+	for _, k := range b1.Names() {
+		v1, ok1 := b1.Get(k)
+		v2, ok2 := b2.Get(k)
 
-	for _, k := range b1.DurationKeys() {
-		v1, _ := b1.Duration(k)
-		v2, _ := b2.Duration(k)
-		if v1 != v2 {
+		if ok1 != ok2 {
 			return false
 		}
-	}
-
-	if len(b1.BytesKeys()) != len(b2.BytesKeys()) {
-		return false
-	}
-
-	for _, k := range b1.BytesKeys() {
-		v1, _ := b1.Bytes(k)
-		v2, _ := b2.Bytes(k)
 
-		if len(v1) != len(v2) {
-			return false
-		}
-
-		for i := 0; i < len(v1); i++ {
-			if v1[i] != v2[i] {
+		switch t1 := v1.(type) {
```

Comment:
Unfortunately not, this only deals with the attributes while bags can contain other fields too.